promql/engine.go

```diff
@@ -1164,8 +1163,9 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, storage.Warnings) {
 	numSteps := int((ev.endTimestamp-ev.startTimestamp)/ev.interval) + 1
 
 	// Create a new span to help investigate inner evaluation performances.
-	span, _ := opentracing.StartSpanFromContext(ev.ctx, stats.InnerEvalTime.SpanOperation()+" eval "+reflect.TypeOf(expr).String())
-	defer span.Finish()
+	ctxWithSpan, span := otel.Tracer("").Start(ev.ctx, stats.InnerEvalTime.SpanOperation()+" eval "+reflect.TypeOf(expr).String())
+	ev.ctx = ctxWithSpan
```

Comment:
I believe the practical difference here is how we are recording the relationship between the internal evaluation spans. Since (depending on the selected `case` based on the expression type) the `eval` function might be called from within itself, we might be starting multiple spans as part of the inner evaluation. Without setting `ev.ctx`, these are recorded as siblings (see the _before_ image for illustration), which is the 'side-effect' of not passing the new context with span. On the contrary, if the `ev.ctx` is updated, the spans will have parent-child relation (see the _after_ image). Since these inner evaluation operations themselves are actually spawning more inner evaluation operations, my take is that parent-child representation is more accurate.

Before
![before-spans](https://user-images.githubusercontent.com/38492574/143787126-c271389d-2500-40c3-b2e1-510c3e4b9fb6.png)

After
![spans-after](https://user-images.githubusercontent.com/38492574/143787133-074d0529-6146-47c5-abd8-a9fc827cba5e.png)

Reply:
if you do 1+1, they should probably not be parent-child.

Reply:
Could you please elaborate on what you mean by the 1+1 example?